libvast/src/system/importer.cpp

```diff
@@ -38,44 +39,73 @@ importer_state::importer_state(caf::event_based_actor* self_ptr)
 }
 
 importer_state::~importer_state() {
-  write_state();
+  write_state(write_mode::with_next);
 }
 
 caf::error importer_state::read_state() {
-  auto file = dir / "next_id";
+  auto file = dir / "current_id_block";
   if (exists(file)) {
     VAST_VERBOSE(self, "reads persistent state from", file);
-    std::ifstream available{to_string(file)};
-    available >> next_id;
-    if (!available.eof())
-      VAST_ERROR(self, "got an invalidly formatted persistence file:", file);
+    std::ifstream state_file{to_string(file)};
+    state_file >> current.end;
```

Comment:
Can you check whether this operation succeeded, e.g., by `if (state_file)`? If this fails already, you should do an hard abort.